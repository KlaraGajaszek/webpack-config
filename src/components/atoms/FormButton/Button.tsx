import React from 'react'
import classnames from 'classnames'

export type ButtonProps = {
  content: string
  bgColor: string
  textColor: string
  onClick?: () => void
  size: string
}

export const Button = ({
  size,
  bgColor,
  textColor,
  content,
  onClick,
}: ButtonProps) => {
  return (
    <button
      onClick={onClick}
      className={classnames(
        `bg-${bgColor} text-${textColor} font-bold py-2 px-4 rounded`,
        {
          'text-xs': size === 'sm',
          'text-xl': size === 'lg',
        }
      )}
    >
      {content}
    </button>
  )
}

export default Button
