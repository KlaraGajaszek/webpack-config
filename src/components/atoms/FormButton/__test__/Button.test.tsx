import React from 'react'
import 'jest-styled-components'
import { screen, render, within } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import renderer from 'react-test-renderer'

import { Button } from '../Button'

describe('Content', () => {
    it('render without crash', () => {
        const mockFunction = jest.fn()
        render(
            <Button
                onClick={mockFunction}
                content="test"
                bgColor="red"
                textColor="red"
                size="sm"
            />
        )
        const formWrapper = screen.getByText('test')
        expect(formWrapper).toBeInTheDocument()
    })
    it('all styles should be render', () => {
        const mockFunction = jest.fn()
        const formButton = renderer
            .create(
                <Button
                    content="test"
                    onClick={mockFunction}
                    bgColor="red"
                    textColor="red"
                    size="sm"
                />
            )
            .toJSON()
        expect(formButton).toMatchSnapshot()
    })
})
