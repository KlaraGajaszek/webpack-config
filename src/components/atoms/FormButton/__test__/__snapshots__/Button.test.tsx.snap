// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content all styles should be render 1`] = `
<button
  className="bg-red text-red font-bold py-2 px-4 rounded text-xs"
  onClick={[MockFunction]}
>
  test
</button>
`;
